src/backend/optimizer/util/clauses.c

```diff
@@ -1868,7 +1868,7 @@ transform_array_Const_to_ArrayExpr(Const *c)
 	if (elemtype == InvalidOid)
 		return (Expr *) c;	/* not an array */
 
-	ac = (ArrayType *) c->constvalue;
+	ac = DatumGetArrayTypeP(c->constvalue);
```

Comment:
I'm ready to approve, but it's probably cleaner to cherry pick `DatumGetArrayTypeP` addition from the commit it was taken, so we not loose the whole history of why it was added.

Reply:
Good point! I have found this commit with fix that I did by code review previously and get it with cherry-pick.
Thank you for attention.